### ISSUE TITLE
Fake MPQ headers

### ISSUE THREAD

**Issue by WaterKnight** [ladislav-zezula/StormLib#300]
Hi,

I have seen no public repo for the MPQ Editor, but since it uses StormLib for parsing MPQs and it fails when loading the following MPQ, it's probably an issue with StormLib:

This Warcraft 3 map makes the MPQ Editor freeze for some couple of seconds, then closes it: https://www.hiveworkshop.com/threads/green-td-pro-deathmatch.319693/

Warcraft 3 (1.36.0.20257) also loads for some time when trying to load the map, but then it can be played.

As can be seen in the first few bytes, a deliberate measure was taken to fool parsers. There are multiple MPQ headers.

MPQ Editor version: 4.0.0.894

**Comment by ladislav-zezula** [comment]
Fixed in commit https://github.com/ladislav-zezula/StormLib/commit/f2c086cef8c9335571f492d9f91f40c1e5f721f5.
MPQ Editor Quick Fix is here: [mpqeditor_en_fix.zip](https://github.com/ladislav-zezula/StormLib/files/11990671/mpqeditor_en_fix.zip)
I'll do a normal update after I run all tests.

**Comment by ZorionMax** [comment]
> Fixed in commit https://github.com/ladislav-zezula/StormLib/commit/f2c086cef8c9335571f492d9f91f40c1e5f721f5.
> 
> MPQ Editor Quick Fix is here: [mpqeditor_en_fix.zip](https://github.com/ladislav-zezula/StormLib/files/11990671/mpqeditor_en_fix.zip)
> 
> I'll do a normal update after I run all tests.

Dear Ladislav Zezula,

I hope this message finds you well. I am facing a problem as I do not have a C++ environment, but I require the StormLib V9.25 dll file. I am aware that you are the developer of StormLib, and therefore, I would like to request a copy of the StormLib V9.25 dll file from you. Your assistance in providing me with this file would be greatly appreciated.

StormLib is an incredibly useful library that plays a vital role in many projects. I believe that StormLib V9.25 is the version I specifically need for my current project.

Thank you once again for your time and assistance. I eagerly await your response.

Best regards,

**Comment by ladislav-zezula** [comment]
I added download entry of StormLib.dll into the [MPQ Downloads](http://www.zezula.net/en/mpq/download.html) section.

**Comment by ZorionMax** [comment]
Dear Mr. Ladislav Zezula,I am writing this letter to express my heartfelt gratitude and sincere appreciation to you. I would like to thank you for your outstanding contributions to the developer community.As a developer myself, I understand that encountering problems and challenges during the software development process is quite common. However, thanks to the StormLib library you have provided, I have been able to easily handle and manage data stored in archive files. The powerful features and user-friendly nature of this library have accelerated my development process and ultimately resulted in a better user experience.I greatly admire your expertise and technical abilities. Your contributions are invaluable to the entire developer community. Whether it is your documentation, sample code, or assistance in addressing user issues, you have demonstrated a commitment to the spirit of open source and mutual support.Furthermore, I would like to express my special thanks for your generosity and selflessness. Your StormLib library is freely available, allowing developers to easily access and use it without any cost. Your generous act sets an example for the developer community and provides us with valuable resources and support.Once again, I want to sincerely thank you for your efforts and dedication. Thank you for providing the StormLib download, which has allowed me to develop my projects more efficiently. Your work holds great significance for me, and it will be forever etched in my memory.Lastly, I would like to express my deepest gratitude to you once again. Your open-source spirit and contributions to the developer community continue to inspire and impact countless individuals. I hope you achieve even greater success in your future endeavors!Thank you sincerely,在 2023年8月7日，01:00，Ladislav Zezula ***@***.***> 写道：﻿
I added download entry of StormLib.dll into the MPQ Downloads section.
